whylog/assistant/regex_assistant/regex.py

```diff
@@ -0,0 +1,119 @@
+"""
+Regex verification and creating (but not finding groups in regex)
+"""
+import re
+
+special_characters = frozenset(['.', '^', '$', '*', '+', '{', '}', '?', '[', ']', '|', '(', ')'])
+
+
+class NoDateGroupError(Exception):
+    def __init__(self, line_content, regex, line_index):
+        self.line_content = line_content
+        self.regex = regex
+        self.line_index = line_index
+
+    def __str__(self):
+        return 'No date group in regex. Regex: %s. Line content: %s' % (
+            self.regex, self.line_content
+        )
+
+
+class NotMatchingRegexError(Exception):
+    def __init__(self, line_content, regex):
+        self.line_content = line_content
+        self.regex = regex
+
+    def __str__(self):
+        return 'Regex do not match line. Regex: %s. Line content: %s' % (
+            self.regex, self.line_content
+        )
+
+
+class DateFromFutureError(Exception):
+    def __init__(self, parsed_date, date_text):
+        self.date_text = date_text
+        self.parsed_date = parsed_date
+
+    def __str__(self):
+        return 'Date is from future. date text %s,  parsed date %s' % (
+            self.date_text, self.parsed_date
+        )
+
+
+def verify_regex(regex, text):
+    """
+    Verifies regex properties such as:
+    - matching a whole text
+    - matching text in a one way only
+    If properties are not met, proper exceptions are returned.
+    """
+    # TODO: verify if regex matches text in a one way only
+
+    # regex must match a whole text from its beginning to end.
+    if not regex[0] == "^":
+        regex = "^" + regex
```

Comment:
Name of this function is 'verify_regex' that mean it should only verify, not also correct. Change the name of the function or implementation.